pkg/framework/plugins/removefailedpods/failedpods_test.go

```diff
@@ -85,9 +88,70 @@ func TestRemoveFailedPods(t *testing.T) {
 				}), nil),
 			},
 		},
+		{
+			description:             "1 container terminated with exitCode 1, 1 eviction",
+			args:                    createRemoveFailedPodsArgs(false, nil, []int32{1}, nil, nil),
+			nodes:                   []*v1.Node{test.BuildTestNode("node1", 2000, 3000, 10, nil)},
+			expectedEvictedPodCount: 1,
+			pods: []*v1.Pod{
+				buildTestPod("p1", "node1", newPodStatus("", "", nil, &v1.ContainerState{
+					Terminated: &v1.ContainerStateTerminated{ExitCode: 1},
+				}), nil),
+			},
+		},
+		{
+			description:             "1 container terminated with exitCode 1, reason NodeAffinity, 1 eviction",
+			args:                    createRemoveFailedPodsArgs(false, nil, []int32{1}, nil, nil),
+			nodes:                   []*v1.Node{test.BuildTestNode("node1", 2000, 3000, 10, nil)},
+			expectedEvictedPodCount: 1,
+			pods: []*v1.Pod{
+				buildTestPod("p1", "node1", newPodStatus("", "", nil, &v1.ContainerState{
+					Terminated: &v1.ContainerStateTerminated{Reason: "NodeAffinity", ExitCode: 1},
+				}), nil),
+			},
+		},
+		{
+			description:             "1 container terminated with exitCode 1, 0 eviction",
+			args:                    createRemoveFailedPodsArgs(false, nil, []int32{2}, nil, nil),
+			nodes:                   []*v1.Node{test.BuildTestNode("node1", 2000, 3000, 10, nil)},
+			expectedEvictedPodCount: 0,
+			pods: []*v1.Pod{
+				buildTestPod("p1", "node1", newPodStatus("", "", nil, &v1.ContainerState{
+					Terminated: &v1.ContainerStateTerminated{ExitCode: 1},
+				}), nil),
+			},
+		},
+		{
+			description:             "2 containers terminated with exitCode 1 and 2 respectively, 2 evictions",
+			args:                    createRemoveFailedPodsArgs(false, nil, []int32{1, 2, 3}, nil, nil),
+			nodes:                   []*v1.Node{test.BuildTestNode("node1", 2000, 3000, 10, nil)},
+			expectedEvictedPodCount: 2,
+			pods: []*v1.Pod{
+				buildTestPod("p1", "node1", newPodStatus("", "", nil, &v1.ContainerState{
+					Terminated: &v1.ContainerStateTerminated{ExitCode: 1},
+				}), nil),
+				buildTestPod("p2", "node1", newPodStatus("", "", nil, &v1.ContainerState{
+					Terminated: &v1.ContainerStateTerminated{ExitCode: 2},
+				}), nil),
+			},
+		},
+		{
+			description:             "2 containers terminated with exitCode 1 and 2 respectively,  1 eviction",
```

Comment:
"2 containers terminated with exitCode 1 and 2 respectively,  1 eviction" -> "2 containers terminated with exitCode 1 and 2 respectively, expected exitCode 1, 1 eviction"